### ISSUE TITLE
defining a new AbstractSparseMatrixCSC requires defining ´_checkbuffers´ (internal function)

### ISSUE THREAD

**Issue by longemen3000** [JuliaLang/julia#43765]
i was in need to define my own type <: `AbstractSparseMatrixCSC` (basically i needed the indexing behaviour of a `SparseMatrixCSC`, but with another storage type). The (not written in the docs) interface for an  `AbstractSparseMatrixCSC` requires:
- `getcolptr`
- `rowvals`
- `nonzeros`
- `size`
the problem is that an internal function `_checkbuffers` is also required, mainly for the display machinery. this function does not require new methods, in fact, is defined in terms of the functions mentioned above (`getrowval` gets dispatched to `rowvals`):

https://github.com/JuliaLang/julia/blob/3b5f98510368d757f468d5ea0674229a27a7c526/stdlib/SparseArrays/src/sparsematrix.jl#L82-L84

`_checkbuffers` should be defined in terms of `AbstractSparseMatrixCSC` instead of the concrete type.

**Comment by stev47** [comment]
I agree, `_goodbuffers` seems to check the assumptions one would expect from `AbstractSparseMatrixCSC`, no need to overspecify.

**Comment by ViralBShah** [comment]
Submit a PR?
